### ISSUE TITLE
Continuously recreating containers

### ISSUE THREAD

**Issue by RobHofmann** [rancher/rancher#19179]
**What kind of request is this (question/bug/enhancement/feature request):**
Question/bug

**Steps to reproduce (least amount of steps as possible):**
Use CentOS 7. Install 3 master nodes, install 4 worker nodes. Look at /var/log/messages on the workers and see a lot of spamming.

**Result:**
/var/log/messages being spammed

**Other details that may be helpful:**

**Environment information**
- Rancher version: `rancher/rancher:v2.1.6`
- Installation option (single install/HA): Single rancher

**Cluster information**
- Cluster type: Custom. On premise CentOS 7 nodes on VMWare
- Machine type: VM (on VMWare ESXi
- CPU's: master nodes have 4 CPU's and worker nodes have 8 CPU's
- Memory: master nodes have 4GB memory and workers have 8GB memory
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.4", GitCommit:"c27b913fddd1a6c480c229191a087698aa92f0b1", GitTreeState:"clean", BuildDate:"2019-02-28T13:37:52Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.6", GitCommit:"b1d75deca493a24a2f87eb1efde1a569e52fc8d9", GitTreeState:"clean", BuildDate:"2018-12-16T04:30:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

cat /var/log/messages
```
Mar 26 11:01:11 xxx01 containerd: time="2019-03-26T11:01:11.306628734+01:00" level=info msg="shim containerd-shim started" address="/containerd-shim/moby/56fde1f6b83bf0e135d2eb0da743c302d0add6a2687b5df13f3de8ebc5a31047/shim.sock" debug=false pid=20167
Mar 26 11:01:11 xx1 kernel: docker0: port 2(vethe69b18a) entered disabled state
Mar 26 11:01:11 xx1 kernel: IPv6: ADDRCONF(NETDEV_CHANGE): vethe69b18a: link becomes ready
Mar 26 11:01:11 xx1 kernel: docker0: port 2(vethe69b18a) entered blocking state
Mar 26 11:01:11 xx1 kernel: docker0: port 2(vethe69b18a) entered forwarding state
Mar 26 11:01:11 xx1 containerd: time="2019-03-26T11:01:11.638563191+01:00" level=info msg="shim reaped" id=56fde1f6b83bf0e135d2eb0da743c302d0add6a2687b5df13f3de8ebc5a31047
Mar 26 11:01:11 xx1 dockerd: time="2019-03-26T11:01:11.648608074+01:00" level=info msg="ignoring event" module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Mar 26 11:01:11 xx1 kernel: docker0: port 2(vethe69b18a) entered disabled state
Mar 26 11:01:11 xx1 kernel: docker0: port 2(vethe69b18a) entered disabled state
Mar 26 11:01:11 xx1 kernel: device vethe69b18a left promiscuous mode
Mar 26 11:01:11 xx1 kernel: docker0: port 2(vethe69b18a) entered disabled state
Mar 26 11:01:11 xx1 kernel: docker0: port 2(veth11de64f) entered blocking state
Mar 26 11:01:11 xx1 kernel: docker0: port 2(veth11de64f) entered disabled state
Mar 26 11:01:11 xx1 kernel: device veth11de64f entered promiscuous mode
Mar 26 11:01:11 xx1 kernel: IPv6: ADDRCONF(NETDEV_UP): veth11de64f: link is not ready
Mar 26 11:01:11 xx1 kernel: docker0: port 2(veth11de64f) entered blocking state
Mar 26 11:01:11 xx1 kernel: docker0: port 2(veth11de64f) entered forwarding state
Mar 26 11:01:11 xx1 containerd: time="2019-03-26T11:01:11.790496191+01:00" level=info msg="shim containerd-shim started" address="/containerd-shim/moby/5ee70b1e70ac2cc06f9ac15f94721342b4dc924b05c9c9af9f77b81c26774fe3/shim.sock" debug=false pid=20257
```

docker events
```
2019-03-26T10:59:07.787493761+01:00 network connect 88e6cf3ef9d087743ff38132f53e0d1b019c7abe2adc10bf6c07cf1f414cd9ff (container=420f628a6f462d9b67ec4b3e78b4ae67ca52b047d15e23e4214f716e1af8b05b, name=none, type=null)
2019-03-26T10:59:07.788983141+01:00 volume mount 58511b2772ed8bd44d8e0f3ea0b0d41d43f6e925fb7e4213f10fcb8749a585c5 (container=420f628a6f462d9b67ec4b3e78b4ae67ca52b047d15e23e4214f716e1af8b05b, destination=/opt/rke-tools, driver=local, propagation=, read/write=true)
2019-03-26T10:59:08.074504233+01:00 container start 420f628a6f462d9b67ec4b3e78b4ae67ca52b047d15e23e4214f716e1af8b05b (image=rancher/rke-tools:v0.1.15, io.rancher.rke.container.name=service-sidekick, maintainer=Rancher Labs <support@rancher.com>, name=service-sidekick)
2019-03-26T10:59:08.091739791+01:00 volume create 5ac90dc9cec628ff01054e31e774f21efab2a267425c97c6c8960243204d7adb (driver=local)
2019-03-26T10:59:08.112689220+01:00 container create f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705 (image=rancher/rke-tools:v0.1.15, maintainer=Rancher Labs <support@rancher.com>, name=rke-log-linker-service-sidekick)
2019-03-26T10:59:08.124433651+01:00 network connect fa1b5b2d1bb4aecf865e5b7fd8d9614e9dc3f3225d29d5034acf998cefc22fab (container=f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705, name=bridge, type=bridge)
2019-03-26T10:59:08.129054124+01:00 volume mount 5ac90dc9cec628ff01054e31e774f21efab2a267425c97c6c8960243204d7adb (container=f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705, destination=/opt/rke-tools, driver=local, propagation=, read/write=true)
2019-03-26T10:59:08.178018828+01:00 container die 420f628a6f462d9b67ec4b3e78b4ae67ca52b047d15e23e4214f716e1af8b05b (exitCode=0, image=rancher/rke-tools:v0.1.15, io.rancher.rke.container.name=service-sidekick, maintainer=Rancher Labs <support@rancher.com>, name=service-sidekick)
2019-03-26T10:59:08.185386968+01:00 network disconnect 88e6cf3ef9d087743ff38132f53e0d1b019c7abe2adc10bf6c07cf1f414cd9ff (container=420f628a6f462d9b67ec4b3e78b4ae67ca52b047d15e23e4214f716e1af8b05b, name=none, type=null)
2019-03-26T10:59:08.205787609+01:00 volume unmount 58511b2772ed8bd44d8e0f3ea0b0d41d43f6e925fb7e4213f10fcb8749a585c5 (container=420f628a6f462d9b67ec4b3e78b4ae67ca52b047d15e23e4214f716e1af8b05b, driver=local)
2019-03-26T10:59:08.390241129+01:00 container start f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705 (image=rancher/rke-tools:v0.1.15, maintainer=Rancher Labs <support@rancher.com>, name=rke-log-linker-service-sidekick)
2019-03-26T10:59:08.414123639+01:00 container kill f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705 (image=rancher/rke-tools:v0.1.15, maintainer=Rancher Labs <support@rancher.com>, name=rke-log-linker-service-sidekick, signal=9)
2019-03-26T10:59:08.481521405+01:00 container die f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705 (exitCode=1, image=rancher/rke-tools:v0.1.15, maintainer=Rancher Labs <support@rancher.com>, name=rke-log-linker-service-sidekick)
2019-03-26T10:59:08.529352461+01:00 network disconnect fa1b5b2d1bb4aecf865e5b7fd8d9614e9dc3f3225d29d5034acf998cefc22fab (container=f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705, name=bridge, type=bridge)
2019-03-26T10:59:08.538726886+01:00 volume unmount 5ac90dc9cec628ff01054e31e774f21efab2a267425c97c6c8960243204d7adb (container=f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705, driver=local)
2019-03-26T10:59:08.550114730+01:00 volume destroy 5ac90dc9cec628ff01054e31e774f21efab2a267425c97c6c8960243204d7adb (driver=local)
2019-03-26T10:59:08.550140693+01:00 container destroy f013d26b154b1615dae9482e73c2ea2849ecc611733dbecb552ec334bf37e705 (image=rancher/rke-tools:v0.1.15, maintainer=Rancher Labs <support@rancher.com>, name=rke-log-linker-service-sidekick)
2019-03-26T10:59:08.592875406+01:00 container create
```

- Docker version: 18.09.3, build 774a1f4

So we installed a fresh cluster using rancher and custom nodes. After installing 3 master nods and 4 worker nodes, we see the behaviour that every 2 minutes the logging is being spammed with the fact that "Veth" interfaces are entering disabled/enabled/blocking states. 

After investigation we saw that some containers are being created & started and deleted right away. We are unable to pinpoint WHY this is happening. Is this expected behaviour? or is this a bug?

Looking forward to your input. Thank you in advance.

**Comment by jgerken** [comment]
We also observe this on our bare metal cluster using Rancher 2.1.7, single server, Debian 9, Docker 18.09.3 build 774a1f4, 1 master (etcd + control plane), 3 workers.

The containers to be killed are

- rke-log-linker-kubelet
- rke-log-linker-kube-proxy
- rke-log-linker-nginx-proxy
- rke-log-linker-service-sidekick

additionally, the container `service-sidekick` dies repeatedly without a matching kill event.

`docker logs service-sidekick` returns an empty log. `docker inspect service-sidekick` returns among the other information:

```json
       "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2019-03-27T20:56:57.889782661Z",
            "FinishedAt": "2019-03-27T20:56:57.913924711Z"
        },
```

this is not surprising considering:

```json
            "Cmd": [
                "/bin/bash"
            ],
            "Entrypoint": null,
```

so the container does actually nothing. Is this really intended? And is it intended that the other containers are killed periodically?

**Comment by billabongrob** [comment]
We're seeing this on Ubuntu 16, Rancher 2.2.2 and Docker 18.9 as well.  It really drives monitoring systems nuts.

**Comment by tontongg** [comment]
We encountered the same problem mentionned by @jgerken : rancher v2.2.4 / 1 master with all three roles and only one worker not on Debian but Ubuntu 18.04.

**Comment by fabiorauber** [comment]
Just bumped into this on Ubuntu 18.04 worker nodes and rancher v2.2.4.

**Comment by soumyalj** [comment]
Reproduced the issue with v2.2.8
Created a cluster with 1 etcd/control node and 3 worker nodes using Ubuntu 16.04 nodes and docker 19.03
Services-sidekick container gets restarted and exits constantly
````
root@ip-172-31-13-255:/home/ubuntu# sudo docker ps -a|grep service-sidekick
4dd6f22c814e        rancher/rke-tools:v0.1.42            "/bin/bash"              26 minutes ago      Exited (0) About a minute ago                       service-sidekick
root@ip-172-31-13-255:/home/ubuntu# sudo docker ps -a|grep service-sidekick
4dd6f22c814e        rancher/rke-tools:v0.1.42            "/bin/bash"              27 minutes ago      Exited (0) 4 seconds ago                          service-sidekick
root@ip-172-31-13-255:/home/ubuntu# sudo docker ps -a|grep service-sidekick
4dd6f22c814e        rancher/rke-tools:v0.1.42            "/bin/bash"              27 minutes ago      Exited (0) 5 seconds ago                          service-sidekick
root@ip-172-31-13-255:/home/ubuntu# sudo docker ps -a|grep service-sidekick
4dd6f22c814e        rancher/rke-tools:v0.1.42            "/bin/bash"              27 minutes ago      Exited (0) 6 seconds ago                          service-sidekick
root@ip-172-31-13-255:/home/ubuntu# sudo docker ps -a|grep service-sidekick
4dd6f22c814e        rancher/rke-tools:v0.1.42            "/bin/bash"              27 minutes ago      Exited (0) 7 seconds ago                          service-sidekick


```
 Repeated logging regarding "Veth" interfaces are entering different states was also observed  in /var/log/syslog

**Comment by pulberg** [comment]
Same issue here, getting around 120k events per day in DD related to -

`rancher/rke-tools` and `rancher/hyperkube`



![Screen Shot 2019-09-09 at 3 16 48 PM](https://user-images.githubusercontent.com/11069351/64563823-e61dd700-d315-11e9-8033-c2388282047d.png)

![Screen Shot 2019-09-09 at 3 17 21 PM](https://user-images.githubusercontent.com/11069351/64563830-eae28b00-d315-11e9-9e14-166828629bd5.png)

**Comment by soumyalj** [comment]
Tested with master-head build from September 9. 
Created a cluster with 1 etcd/control node and 3 worker nodes using Ubuntu 16.04 nodes and docker 19.03
Services-sidekick container does not restart constantly
```
91c78fc58405        rancher/rke-tools:v0.1.47            "/bin/bash"              29 minutes ago      Created                                          service-sidekick
```
```
root@ip-172-31-3-151:/home/ubuntu# grep -i start /var/log/syslog | egrep "kube-proxy"
root@ip-172-31-3-151:/home/ubuntu# grep -i start /var/log/syslog | egrep "service-sidekick"

```
Repeated logging regarding "Veth" interfaces entering different states was not observed  in /var/log/syslog . Sample logs below from /var/log/syslog
```




Sep  9 20:31:55 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:55.708069646Z" level=debug msg="Calling GET /containers/json?filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:55 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:55.715611934Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%2C%22status%22%3A%7B%22running%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:55 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:55.782129169Z" level=debug msg="Calling GET /images/json"
Sep  9 20:31:55 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:55.810096211Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:55 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:55.812915390Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:55 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:55.815475747Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:55 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:55.960303887Z" level=debug msg="Calling GET /version"
Sep  9 20:31:56 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:56.079471736Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:56 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:56.081577718Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:56 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:56.105760578Z" level=debug msg="Calling GET /version"
Sep  9 20:31:56 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:56.106302287Z" level=debug msg="Calling GET /version"
Sep  9 20:31:56 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:56.106726982Z" level=debug msg="Calling GET /version"
Sep  9 20:31:56 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:56.253578384Z" level=debug msg="Calling GET /images/json"
Sep  9 20:31:57 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:57.084598936Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:57 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:57.086777680Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:57 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:57.707886196Z" level=debug msg="Calling GET /containers/json?filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:57 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:57.709506554Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%2C%22status%22%3A%7B%22running%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:57 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:57.710733866Z" level=debug msg="Calling GET /containers/json?filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:57 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:57.712094648Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%2C%22status%22%3A%7B%22running%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:58 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:58.089820722Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:58 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:58.092303859Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:59 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:59.095251523Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:59 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:59.097705932Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:59 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:59.707950214Z" level=debug msg="Calling GET /containers/json?filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:59 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:59.709589397Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%2C%22status%22%3A%7B%22running%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:59 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:59.710871497Z" level=debug msg="Calling GET /containers/json?filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:31:59 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:31:59.712281646Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%2C%22status%22%3A%7B%22running%22%3Atrue%7D%7D&limit=0"
Sep  9 20:32:00 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:32:00.100959562Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:32:00 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:32:00.103328540Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:32:00 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:32:00.961337150Z" level=debug msg="Calling GET /version"
Sep  9 20:32:01 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:32:01.106447922Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"
Sep  9 20:32:01 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:32:01.108576662Z" level=debug msg="Calling GET /containers/json?all=1&filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dcontainer%22%3Atrue%7D%7D&limit=0"
Sep  9 20:32:01 ip-172-31-3-151 dockerd[14076]: time="2019-09-09T20:32:01.711278465Z" level=debug msg="Calling GET /containers/json?filters=%7B%22label%22%3A%7B%22io.kubernetes.docker.type%3Dpodsandbox%22%3Atrue%7D%7D&limit=0"



```

**Comment by zeroum** [comment]
@soumyalj Which version of Rancher contains rancher/rke-tools >= v0.1.47 ??

I can't find the bug fix in v2.2.9's release notes.

**Comment by sangeethah** [comment]
@zeroum This is fix is made in 2.3.0 release.
